app/models/pydantic/datamart.py

```
@@ -1,12 +1,68 @@
 from enum import Enum
-from typing import Dict, Optional
+from typing import Dict, Literal, Optional, Union
 from uuid import UUID
 
-from pydantic import Field
+from pydantic import Field, root_validator, validator
 
 from app.models.pydantic.responses import Response
 
 from .base import StrictBaseModel
+from ...crud.geostore import get_gadm_geostore_id
+
+
+class AreaOfInterest(StrictBaseModel):
+    async def get_geostore_id(self) -> UUID:
```

Maybe add '@abstractmethod' here (from abc package), since this has to be overridden (since not returning anything).

Or maybe throw an exception if this method is actually called? I see that Global() does not actually define a get_geostore_id method, so should maybe throw an exception if global.get_geostore_id() is called due to a bug.

---

I removed the `abstractmethod` since this isn't required for child classes anymore. As a compromise I have the base method return 'not implemented' error so child classes that don't implement it won't have to worry about it.